src/macro-expansion.md

```diff
@@ -1 +1,154 @@
 # Macro expansion
+
+Macro expansion happens during parsing. `rustc` has two parsers, in fact: the
+normal Rust parser, and the macro parser. During the parsing phase, the normal
+Rust parser will call into the macro parser when it encounters a macro
+definition or macro invocation (TODO: verify). The macro parser, in turn, may
+call back out to the Rust parser when it needs to bind a metavariable (e.g.
+`$my_expr`) while parsing the contents of a macro invocation. The code for macro
+expansion is in [`src/libsyntax/ext/tt/`][code_dir]. This chapter aims to
+explain how macro expansion works.
+
+### Example
+
+It's helpful to have an example to refer to. For the remainder of this chapter,
+whenever we refer to the "example _definition_", we mean the following:
+
+```rust
+macro_rules! printer {
+    (print $mvar:ident) => {
+        println!("{}", $mvar);
+    }
+    (print twice $mvar:ident) => {
+        println!("{}", $mvar);
+        println!("{}", $mvar);
+    }
+}
+```
+
+`$mvar` is called a _metavariable_. Unlike normal variables, rather than binding
+to a value in a computation, a metavariable binds _at compile time_ to a tree of
+_tokens_. A _token_ zero or more symbols that together have some meaning. For
```

Comment:
> A _token_ zero or more symbols that together have some meaning.

This sentence is not grammatical and I'm not quite sure how to fix it. =) In particular, I don't think of a token as "zero or more symbols" (and it's sort of unclear to me what you mean by symbol, which in parsing terminology is often used to mean the union of token and nonterminal).

I think I would maybe say something like this:

"A _token_ is a single "unit" of the grammar, such as an identifier (e.g., `print`) or punctuation (e.g., `=>`). Token trees resulting from paired parentheses-like characters (`(...)`, `[...]`, and `{...}`) -- they include the open and close and all the tokens in between (we do require that parentheses-like characters be balanced)."

but it doesn't seem like the best either :)